pl_bolts/datamodules/base_datamodule.py

```diff
@@ -0,0 +1,120 @@
+import os
+from typing import Optional, Tuple
+
+import torch
+from pytorch_lightning import LightningDataModule
+from torch.utils.data import DataLoader, random_split
+
+from pl_bolts.utils.warnings import warn_missing_pkg
+
+try:
+    from torchvision import transforms as transform_lib
+except ModuleNotFoundError:
+    warn_missing_pkg("torchvision")  # pragma: no-cover
+    _TORCHVISION_AVAILABLE = False
+else:
+    _TORCHVISION_AVAILABLE = True
+
+
+class BaseDataModule(LightningDataModule):
+
+    extra_args = {}
+
+    def __init__(
+        self,
+        dataset_cls,
+        dims: Tuple[int, int, int],
+        data_dir: Optional[str] = None,
+        val_split: int = 5000,
```

Comment:
Should this be an integer, a float, or perhaps optionally both? I think intuitively a float makes most sense, as you wont have to hard-code for future datasets.

but this is fine as-is too.